File: TypeScript/509. Fibonacci Number.ts
function fib(n: number): number {
    var fib: Array<number> = [0, 1];
    for(var i = 2; i <= n; i++){
        fib[i] = fib[i - 1] + fib[i - 2];
    }
    return fib[n];
};

File: C++/509. Fibonacci Number.cpp
class Solution {
public:
    int fib(int n) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);

        int f[n + 2];
        f[0] = 0; f[1] = 1;

        for (int i = 2; i <= n; i++) f[i] = f[i - 1] + f[i - 2];

        return f[n];
    }
};
